app/utils/PhoneNumberValidation.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+import play.api.data.Forms.text
+
+object PhoneNumberValidation {
+
+  val phoneNumberRegex =
```

Comment:
for such a complex regex we should have unit tests that test all the possible forms

Reply:
pr has been updated

Reply:
I think it would be useful to have a comment block above this regex to list the valid formats (one example of each - it looks like there are 7 alternatives - separated by `|`)